### ISSUE TITLE
Use add_listener function instead of on_socket_response

### ISSUE THREAD

**Issue by burtonwilliamt** [kiki7000/discord.py-components#45]
In this line, I think the intention is add a socket_response listener
https://github.com/kiki7000/discord.py-components/blob/49dafb8f5efa84c655649608b7ffd157b86d0040/discord_components/client.py#L86

However, an unwanted side effect of this way of adding a listener is that it overwrites any other on_socket_response listener function already defined on the bot. I think a better solution would be to use discord.py's [Bot.add_listener](https://github.com/Rapptz/discord.py/blob/f56543df15a17434689542f227f466608739e865/discord/ext/commands/bot.py#L397) function:

```python
self.bot.add_listener(on_socket_response, name='socket_response')
```

This listener will still be called even if some other library/user adds 'socket_response' listeners, and also won't replace existing on_socket_response methods.

**Comment by Shaqalito** [comment]
> In this line, I think the intention is add a socket_response listener
> https://github.com/kiki7000/discord.py-components/blob/49dafb8f5efa84c655649608b7ffd157b86d0040/discord_components/client.py#L86
> 
> However, an unwanted side effect of this way of adding a listener is that it overwrites any other on_socket_response listener function already defined on the bot. I think a better solution would be to use discord.py's [Bot.add_listener](https://github.com/Rapptz/discord.py/blob/f56543df15a17434689542f227f466608739e865/discord/ext/commands/bot.py#L397) function:
> 
> ```python
> self.bot.add_listener(on_socket_response, name='socket_response')
> ```
> 
> This listener will still be called even if some other library/user adds 'socket_response' listeners, and also won't replace existing on_socket_response methods.

Fantastic solution to a problem I've passed the passt 7 and a half hours on !

**Comment by kiki7000** [comment]
That would be nice
Thank you for the suggestion

**Comment by kiki7000** [comment]
But what if the bot is type client?

**Comment by Shaqalito** [comment]
Oof I don't know...

**Comment by kiki7000** [comment]
Implemented on 0.5.2.4
